lwrunner/awsrunner.go

```diff
@@ -0,0 +1,141 @@
+//
+// Author:: Nicholas Schmeller (<nick.schmeller@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package lwrunner
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2instanceconnect"
+	"golang.org/x/crypto/ssh"
+)
+
+type AWSRunner struct {
+	Runner           Runner
+	Region           string
+	AvailabilityZone string
+	InstanceID       string
+}
+
+func NewAWSRunner(amiImageId, host, region, availabilityZone, instanceID string, callback ssh.HostKeyCallback) (*AWSRunner, error) {
+	// Look up the AMI name of the runner
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	svc := ec2.New(ec2.Options{
+		Credentials: cfg.Credentials,
+		Region:      region,
+	})
+	input := ec2.DescribeImagesInput{
+		ImageIds: []string{
+			amiImageId,
+		},
+	}
+	result, err := svc.DescribeImages(context.Background(), &input)
+	if err != nil {
+		return nil, err
+	}
+	if len(result.Images) != 1 {
+		return nil, fmt.Errorf("expected to find only one AMI")
+	}
+
+	// Heuristically assign SSH username based on AMI name
```

Comment:
Note: allow user to supply username as flag for EC2IC. Give user the power to override the heuristic